src/main/native/compression/IntelDeflater.cc

```diff
@@ -243,8 +281,8 @@ JNIEXPORT jint JNICALL Java_com_intel_gkl_compression_IntelDeflater_deflateNativ
 
 
     // release buffers
-    env->ReleasePrimitiveArrayCritical(inputBuffer, next_in, 0);
     env->ReleasePrimitiveArrayCritical(outputBuffer, next_out, 0);
+    env->ReleasePrimitiveArrayCritical(inputBuffer, next_in, 0);
```

Comment:
??

Reply:
I removed this as discussed. This can potentially be moved to a separate PR.